File: mirai-core/src/main/java/net/mamoe/mirai/task/MiraiThreadPool.java
package net.mamoe.mirai.task;

import java.io.Closeable;
import java.util.concurrent.ScheduledThreadPoolExecutor;

/**
 * @author NaturalHG
 */
public final class MiraiThreadPool extends ScheduledThreadPoolExecutor implements Closeable {
    private static MiraiThreadPool instance = new MiraiThreadPool();

    public static MiraiThreadPool getInstance() {
        return instance;
    }

    MiraiThreadPool() {
        super(10);

        /*super(0,
                Integer.MAX_VALUE,
                60L,
                TimeUnit.SECONDS,
                new SynchronousQueue<>()
        );*/
    }


    @Override
    public void close(){
        this.shutdown();
        if(!this.isShutdown()){
            this.shutdownNow();
        }
    }

}

File: mirai-core/src/jvmMain/kotlin/net/mamoe/mirai/task/MiraiThreadPool.kt
package net.mamoe.mirai.task

import net.mamoe.mirai.Mirai

import java.io.Closeable
import java.util.concurrent.ScheduledThreadPoolExecutor

/**
 * @author NaturalHG
 */
class MiraiThreadPool internal constructor()/*super(0,
                Integer.MAX_VALUE,
                60L,
                TimeUnit.SECONDS,
                new SynchronousQueue<>()
        );*/ : ScheduledThreadPoolExecutor(10), Closeable {


    override fun close() {
        this.shutdown()
        if (!this.isShutdown) {
            this.shutdownNow()
        }
    }

    companion object {
        val instance = MiraiThreadPool()

        @JvmStatic
        fun main(args: Array<String>) {
            println(Mirai.VERSION)
        }
    }

}
